src/main/java/bio/terra/cli/command/resource/Resolve.java

```diff
@@ -42,35 +49,52 @@ public class Resolve extends BaseCommand {
   @Override
   protected void execute() {
     workspaceOption.overrideIfSpecified();
-    Resource resource = Context.requireWorkspace().getResource(resourceNameOption.name);
+    String[] paths = resourceName.split("/");
+    if (paths.length > 2) {
+      throw new UserActionableException(
+          "Invalid path provided, only support resolving [resource name] or"
+              + "[data source name]/[resource name]");
+    }
 
-    String cloudId;
+    Resource resource = Context.requireWorkspace().getResource(paths[0]);
+
+    JSONObject cloudIds = new JSONObject();
     switch (resource.getResourceType()) {
       case GCS_BUCKET:
-        cloudId = ((GcsBucket) resource).resolve(!excludeBucketPrefix);
+        cloudIds.put(resource.getName(), ((GcsBucket) resource).resolve(!excludeBucketPrefix));
         break;
       case GCS_OBJECT:
-        cloudId = ((GcsObject) resource).resolve(!excludeBucketPrefix);
+        cloudIds.put(resource.getName(), ((GcsObject) resource).resolve(!excludeBucketPrefix));
         break;
       case BQ_DATASET:
-        cloudId = ((BqDataset) resource).resolve(bqPathFormat);
+        cloudIds.put(resource.getName(), ((BqDataset) resource).resolve(bqPathFormat));
         break;
       case BQ_TABLE:
-        cloudId = ((BqTable) resource).resolve(bqPathFormat);
+        cloudIds.put(resource.getName(), ((BqTable) resource).resolve(bqPathFormat));
+        break;
+      case DATA_SOURCE:
+        if (paths.length == 2) {
+          cloudIds.put(paths[1], ((DataSource) resource).resolve(paths[1]));
+        } else {
+          var resources = ((DataSource) resource).getDataSourceWorkspace().getResources();
+          resources.stream()
+              .filter(r -> r.getResourceType() != Resource.Type.DATA_SOURCE)
+              .forEach(r -> cloudIds.put(r.getName(), r.resolve()));
+        }
         break;
       default:
-        cloudId = resource.resolve();
+        cloudIds.put(resource.getName(), resource.resolve());
     }
-    JSONObject object = new JSONObject();
-    object.put(resource.getName(), cloudId);
-    formatOption.printReturnValue(object, this::printText, this::printJson);
+    formatOption.printReturnValue(cloudIds, this::printText, this::printJson);
   }
 
   private void printText(JSONObject object) {
-    OUT.println(object.get(resourceNameOption.name));
+    for (var key : object.keySet()) {
+      OUT.println(key + ": " + object.get((String) key));
+    }
   }
 
   private void printJson(JSONObject object) {
-    OUT.println(object);
+    OUT.println(object.toString(2));
```

Comment:
nit: why 2 here?

Reply:
without indentation: it looks like this: 
```
{"newbucket":"gs://terra-a7f6c39b-32b7-439a-8277-957b9ca29b78-bucket","newdataset":"terra-vdevel-silken-lime-2975.newdataset","testtest":"gs://terra-40aa1f27-046f-4f85-a918-874956a32df2-bucket"}
```
this is the indentation of terra resource list --format=json
```
[ {
  "id" : "1f23c921-172b-4fab-85ad-b53cfde86b89",
  "name" : "cloudbuildworkspace",
  "description" : null,
  "resourceType" : "DATA_SOURCE",
  "stewardshipType" : "REFERENCED",
  "cloningInstructions" : "COPY_NOTHING",
  "accessScope" : null,
  "managedBy" : null,
  "privateUserName" : null,
  "privateUserRoles" : null,
  "dataSourceWorkspaceUuid" : "11d8309e-7b53-47c8-a363-86c0c32e91ac",
  "title" : "cloudbuoldworkspace",
  "shortDescription" : "",
  "version" : ""
}, {
  "id" : "8fdaa72f-f428-43c8-94d7-3f17e8d0e94a",
  "name" : "demonotebook",
  "description" : null,
  "resourceType" : "AI_NOTEBOOK",
  "stewardshipType" : "CONTROLLED",
  "cloningInstructions" : "COPY_NOTHING",
  "accessScope" : "PRIVATE_ACCESS",
  "managedBy" : "USER",
  "privateUserName" : "yuhuyoyo@google.com",
  "privateUserRoles" : null,
  "projectId" : "terra-vdevel-potent-beet-7262",
  "instanceId" : "yuhuyoyo-20220629-210851",
  "location" : "us-central1-a",
  "instanceName" : "projects/terra-vdevel-potent-beet-7262/locations/us-central1-a/instances/yuhuyoyo-20220629-210851",
  "state" : "ACTIVE",
  "metadata" : {
    "container" : "us-central1-docker.pkg.dev/terra-vdevel-potent-beet-7262/my-repo/rcpu-terra",
    "proxy-mode" : "service_account",
    "shutdown-script" : "/opt/deeplearning/bin/shutdown_script.sh",
    "notebooks-api" : "PROD",
    "enable-guest-attributes" : "TRUE",
    "version" : "93",
    "post-startup-script" : "https://raw.githubusercontent.com/DataBiosphere/terra-workspace-manager/main/service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh",
    "terra-workspace-id" : "dockerdemoworkspace",
    "framework" : "Container: R",
    "restriction" : "",
    "proxy-url" : "14318b060e71dc48-dot-us-central1.notebooks.googleusercontent.com",
    "title" : "Base.Container.GPU",
    "terra-cli-server" : "verily-devel"
  },
  "proxyUri" : "14318b060e71dc48-dot-us-central1.notebooks.googleusercontent.com",
  "createTime" : "2022-06-29T21:08:52.837402148Z"
} ]
```
It is indented by 2.